src/WinRT.Runtime/Marshalers.cs

```diff
@@ -668,7 +668,8 @@ static MarshalGeneric()
                 DisposeMarshaler = ABI.System.NonBlittableMarshallingStubs.NoOpFunc;
                 DisposeAbi = ABI.System.NonBlittableMarshallingStubs.NoOpFunc;
             }
-            else if (typeof(T).IsValueType)
+            else if (typeof(T).IsValueType ||

```

Comment:
I'm worried this and the path below might regress size, because now this entire branch must be preserved for all reference types as well (the `Exception` check can't be folded), whereas previously with just `IsValueType` that could be handled by the importer directly, as it's an intrinsic, and by the linker too 🤔

Reply:
> the `Exception` check can't be folded

While it can be folded by the JIT so I don't think it would result in size regression. https://godbolt.org/z/5nKG7MWon

Reply:
The point is this whole path could be removed entirely for all reference types, but now it must be kept in all cases.

Reply:
Latest changes look to address it.  We no longer have a separate generic instantiation for each class, but just for `canon` and `exception`.  Before these changes we did.  With the latest, we just have the below:

![image](https://github.com/user-attachments/assets/635ef80f-b6b1-4e15-935e-9f6c397fcb8d)